pennylane_lightning/core/src/simulators/lightning_qubit/gates/cpu_kernels/GateImplementationsLM.hpp

```diff
@@ -1898,16 +1961,15 @@ class GateImplementationsLM : public PauliGenerator<GateImplementationsLM> {
             wires_parity |=
                 (static_cast<std::size_t>(1U) << (num_qubits - wire - 1));
         }
-        auto core_function =
-            [wires_parity, &shifts](
-                std::complex<PrecisionT> *arr,
-                const std::vector<std::size_t> &indices,
-                [[maybe_unused]] const std::vector<std::complex<PrecisionT>>
-                    &coeffs_in) {
-                for (const auto &k : indices) {
-                    arr[k] *= shifts[std::popcount(k & wires_parity) % 2];
-                }
-            };
+        auto core_function = [wires_parity,
+                              &shifts](std::complex<PrecisionT> *arr,
+                                       const std::vector<std::size_t> &indices,
+                                       const std::size_t offset) {
+            for (auto k : indices) {
```

Comment:
```suggestion
            for (auto &k : indices) {
```
Better to use it as a reference

Reply:
I think it is better copying POD, don't you think?

Reply:
No, because it does not change, it just gets the value beside copies the value takes more computing time :thinking: ****

Reply:
It probably compiles to the same assembly under optimization, but a debug build or if I'd wrote assembly naively by hand I would: 
- reference: copy the the `k` address (itself an `int`) into an `int` register, copy the integer at that address into an `int` register, do my thing.
- copy: copy the integer into an `int` register, do my thing.
In a loop over an array, I also don't know whether the next address will hold a value which is just +8bytes in the first case, but I'm absolutely sure of it in the second case.

Reply:
Prefer copies for POD is the way to go :+1:

Reply:
This is actually in the [C++ dev guide](https://www.notion.so/xanaduai/C-Development-Guide-d5b53354ca844e2d8780a78c41302ce2?pvs=4#cf671dc822da4b1eb5163628e643aeeb).